src/axom/core/utilities/BitUtilities.hpp

```diff
@@ -85,7 +87,7 @@ struct BitTraits<std::uint8_t>
 AXOM_HOST_DEVICE inline int trailingZeros(std::uint64_t word)
 {
   /* clang-format off */
-#if defined(AXOM_DEVICE_CODE) && defined(AXOM_USE_GPU)
+#if defined(__CUDA_ARCH__) && defined(AXOM_USE_CUDA)
```

Comment:
Is the intent here to suggest that we need to maintain two different versions of code like this, one for CUDA and one for HIP?

The second check (if AXOM_USE_CUDA is defined should not be needed, because you added the check for that at line 26 above.